### ISSUE TITLE
How set on lockscreen? 

### ISSUE THREAD

**Issue by dancho12** [Marcel1202/HTMLWallpaper#6]
Hello! Can i set this plugin on lock-screen? When i'am try have crush. Tell me it is impossible?

**Comment by redbrain** [comment]
I'm also interested in this feature (if it is possible)

**Comment by redbrain** [comment]
I looked into doing this. Apparently the login screen is not a plasma service, but rather an sddm theme. Since both use QML it may not be difficult to convert this plugin into a login theme, but I lack the knowledge to do so.

**Comment by Marcel1202** [comment]
The lock screen and login screen are two different things. The lock screen appears after idling a set amount of time or after clicking "Lock" in plasmashell. It is provided by plasma. The login screen on the other hand is provided by SDDM. That's the screen you see when no user is currently logged in (e.g. when freshly booting up your system). This issue refers to the lock screen.

**Comment by renyuneyun** [comment]
Interestingly, in recent versions of Plasma (I'm currently on 5.21.3, Archlinux), this plugin can be used directly for lock screen.
Displaying the webpage works without issues. (Though in a previous version of Plasma/KDE Framework, it was a black screen.)

However, there is an issue with **unlocking**. After inputting password and hit enter, it returns to the lock screen. The workaround I found is to select "switch user", and enter password there.

**Comment by redbrain** [comment]
That's good news for the plugin.
If you think the unlock process is an issue with KDE, you may want to file a bug with them.

**Comment by renyuneyun** [comment]
@redbrain I think the unlock process is an issue with this plugin (or a bad design in KDE, but not a *bug*). Other (built-in) wallpaper types/plugins seem to work fine.

Though, apparently, because the setting page for lock screen does not allow you to install plugins, it seems Plasma is not currently designed to allow third-party plugins to be used there. Not sure what this implies regarding future updates.

**Comment by Marcel1202** [comment]
Logs show a segmentation fault when the login succeeds:
`kernel	kscreenlocker_g[9609]: segfault at ... error 4 in libQt5WebEngine.so.5.15.3[...]`
My somewhat uneducated guess is that when KScreenLocker closes to open the user session it causes the plugin to crash instead of exiting smoothly, taking KScreenLocker with it in the process. Then Plasma detects the crash of KScreenLocker and restarts it. 
[This plugin](https://github.com/catsout/wallpaper-engine-kde-plugin) manages to run QtWebEngine in the lockscreen without any crashes so we may be able to take inspiration from there.

**Comment by N-Upchurch** [comment]
With Plasma 5, this plugin *did* work for me on the lock screen, but since upgrading to plasma 6, I get the old "The screen locker is broken and unlocking is not possible anymore" message immediately after locking the screen both on Arch/Endeavour and Fedora Kinoite Rawhide. Is anyone able to advise whether this is a bug in Plasma, HTMLWallpaper, or is this a known issue since it was ported to Plasma 6 and lock screen support is now essentially a feature request? I ask because I'd like to make a bug report on bugs.kde.org if this is an issue with Plasma or KScreenLocker.

**Comment by renyuneyun** [comment]
@N-Upchurch Haven't tried, but I would guess this is related to the fact that Plasma 6 changed APIs, and made all existing plugins and widgets/plasmoids unusable unless explicitly migrated and expressed compatible with Plasma 6.
(I understand the need to clean up old APIs, but this is quite rough...)

**Comment by N-Upchurch** [comment]
> Plasma 6 changed APIs, and made all existing plugins and widgets/plasmoids unusable unless explicitly migrated

I am using the Plasma 6 version of HTMLWallpaper.

**Comment by akeyx** [comment]
Yeah, this sadly stopped working for me on Plasma 6 lockscreen ;(  .
So is there a hope for update to Plasma 6 compatibility in lockscreen mode?

**Comment by kelvie** [comment]
On plasma 6, it crashes due to this error (in `journalctl -u user@1000.service -b 0`).

```
 kscreenlocker_greet[303695]: QtWebEngineQuick::initialize() must be called from the Qt gui thread.
```

**Comment by MarcelRichter-GitHub** [comment]
> Is anyone able to advise whether this is a bug in Plasma, HTMLWallpaper, or is this a known issue since it was ported to Plasma 6 and lock screen support is now essentially a feature request? I ask because I'd like to make a bug report on bugs.kde.org if this is an issue with Plasma or KScreenLocker.

To clarify, HTMLWallpaper has always crashed on the Arch Linux Plasma 5 I used back then; hence lock screen support is labelled as a feature request here.

> On plasma 6, it crashes due to this error (in `journalctl -u user@1000.service -b 0`).
> 
> ```
>  kscreenlocker_greet[303695]: QtWebEngineQuick::initialize() must be called from the Qt gui thread.
> ```

If this is really the error causing the crash, then lock screen might be impossible. This is the same error message that causes #2, which to my knowledge, other QtWebEngine plugins also weren't able to solve.

If someone knows how to make it happen, pull request are, of course, very welcome.
